### PR TITLE
Failed sends

### PR DESCRIPTION
**Pull request by hkalodner**
- Add test to verify incorrect behavior of triggering sends even when transaction reverts (Fixed with arbos update)
- Update ArbOS with https://github.com/OffchainLabs/arb-os/pull/193

### CODE REVIEW

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 12
           

Complexity increasing per file
==============================
- packages/arb-tx-aggregator/arbostest/calls_test.go  2
- packages/arb-tx-aggregator/arbostestcontracts/failedsend.go  3
- packages/arb-tx-aggregator/arbostest/nested_send_test.go  8
         

Clones added
============
- packages/arb-tx-aggregator/arbostestcontracts/failedsend.go  4
         

Clones removed
==============
+ packages/arb-tx-aggregator/arbostest/l2_test.go  -1
+ packages/arb-tx-aggregator/arbostest/arbos_test.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=6057380&bid=19842524)

File: packages/arb-tx-aggregator/arbostestcontracts/failedsend.sol [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity ^0.5.17;
+
+interface Sys {
+    // Send given amount of Eth to dest with from sender.
```
Comment (edfelten) [inline packages/arb-tx-aggregator/arbostestcontracts/failedsend.sol:22]: I can't parse the comment "... to dest with from sender".

Comment (hkalodner) [reply packages/arb-tx-aggregator/arbostestcontracts/failedsend.sol]: Fixed the comment

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 12
           

Complexity increasing per file
==============================
- packages/arb-tx-aggregator/arbostestcontracts/failedsend.go  3
- packages/arb-tx-aggregator/arbostest/nested_send_test.go  8
- packages/arb-tx-aggregator/arbostest/calls_test.go  2
         

Clones added
============
- packages/arb-tx-aggregator/arbostestcontracts/failedsend.go  4
         

Clones removed
==============
+ packages/arb-tx-aggregator/arbostest/arbos_test.go  -1
+ packages/arb-tx-aggregator/arbostest/l2_test.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=6057380&bid=19842524)

**Pull request merged**